File: app/src/main/java/ru/nehodov/weatherforecast/dao/HourlyDao.java
package ru.nehodov.weatherforecast.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ru.nehodov.weatherforecast.entities.Hourly;

@Dao
public interface HourlyDao {

    @Insert
    void insertHourlyForecast(Hourly[] hourlies);

    @Query("SELECT * FROM hourly")
    LiveData<List<Hourly>> getHourlyForecast();

    @Query("DELETE FROM hourly")
    void deleteHourlyForecast();
}

File: app/src/main/java/ru/nehodov/weatherforecast/dao/HourlyDao.kt
package ru.nehodov.weatherforecast.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import ru.nehodov.weatherforecast.entities.Hourly

@Dao
interface HourlyDao {

    @Insert
    fun insert(hourlies: List<Hourly>)

    @Query("SELECT * FROM hourly")
    fun hourlyData(): List<Hourly>

    @Query("DELETE FROM hourly")
    fun deleteAll()

}